backend/app/usecase/validator/violation.go

```diff
@@ -0,0 +1,25 @@
+package validator
+
+// Violation represents a type of invalid error encountered by the validator.
+type Violation int
+
+const (
+	Valid Violation = iota
+	InvalidAlias
+	InvalidLongLink
+	AliasTooLong
+	LongLinkTooLong
+	HasFragmentCharacter
+)
+
+// Returns a string representation of the violation.
+func (v Violation) String() string {
+	return [...]string{
+		"Valid",
+		"InvalidAlias",
+		"InvalidLongLink",
+		"AliasTooLong",
+		"LongLinkTooLong",
+		"HasFragmentCharacter",
+	}[v]
+}
```

Comment:
You can also change line 4 to `type Violation string`

Reply:
Forgot to address this. 🙈 Thanks for reminding me!